src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -2529,6 +2531,28 @@ public virtual void AttachTo(string entitySetName, object entity, string etag)
             this.entityTracker.InternalAttachEntityDescriptor(descriptor, true /*failIfDuplicated*/);
         }
 
+        /// <summary>Changes the state of the specified object to be deleted in the <see cref="Microsoft.OData.Client.DataServiceContext" />.</summary>
+        /// <param name="entity">The tracked entity to be changed to the deleted state.</param>
+        /// <param name="dependsOnIds">DependsOnIds for the deleted entity.</param>
+        /// <exception cref="System.ArgumentNullException">When <paramref name="entity" /> is null.</exception>
+        /// <exception cref="System.InvalidOperationException">When the object is not being tracked by the <see cref="Microsoft.OData.Client.DataServiceContext" />.</exception>
+        /// <remarks>
+        /// Existing objects in the Added state become detached.
+        /// </remarks>
+        public virtual void DeleteObject(object entity, params object[] dependsOnIds)
+        {
+            var dependsOn = new List<string>();
+            if (dependsOnIds != null)
+            {
+                foreach (var id in dependsOnIds)
+                {
+                    dependsOn.Add(id.ToString());

```

Comment:
[Question] Are we always sure that the depends on can always be converted to a string? What types are usually used for this, Guids? ints?

Reply:
This is the basic assumption since `ODataBatchWriter.CreateOperationRequestMessage` only accepts `IEnumerable<string>` as the dependsOnIds.
https://github.com/OData/odata.net/blob/25ea344714d291d169b48ebf71c09721014989aa/src/Microsoft.OData.Core/Batch/ODataBatchWriter.cs#L294-L295